data env blocks without default throw an error

this doesn't work
```
data env version {
  key     = "CI_PIPELINE_VERSION"
}
```
```
INFO[0000] togomak (version=1.2.1)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x196f888]

goroutine 27 [running]:
github.com/srevinsaju/togomak/v1/pkg/blocks/data.(*EnvProvider).DecodeBody(0xc00134a180, {0x20d01f8, 0xc00148a0b0})
	/home/runner/work/togomak/togomak/pkg/blocks/data/env.go:98 +0x2e8
github.com/srevinsaju/togomak/v1/pkg/ci.Data.Run({{0xc00146bc04, 0x3}, {0xc00146bc07, 0x7}, {0x0, 0x0}, {0x0, 0x0}, {0x20d01f8, 0xc00148a0b0}}, ...)
	/home/runner/work/togomak/togomak/pkg/ci/data_run.go:43 +0x30a
github.com/srevinsaju/togomak/v1/pkg/orchestra.Orchestra.func1({0xc001485080, 0x10})
	/home/runner/work/togomak/togomak/pkg/orchestra/orchestra.go:306 +0xce
created by github.com/srevinsaju/togomak/v1/pkg/orchestra.Orchestra
	/home/runner/work/togomak/togomak/pkg/orchestra/orchestra.go:305 +0x1f3e
```

if i add 
```
default = ""
```
it works

--- a/examples/env/togomak.hcl
+++ b/examples/env/togomak.hcl
@@ -1,0 +1,12 @@
+togomak {
+  version = 1
+}
+
+data "env" "HOME" {
+  key = "HOME"
+}
+
+stage "example" {
+  name   = "example"
+  script = "echo ${data.env.HOME.value}"
+}

--- a/pkg/blocks/data/env.go
+++ b/pkg/blocks/data/env.go
@@ -94,7 +94,11 @@ func (e *EnvProvider) DecodeBody(body hcl.Body) hcl.Diagnostics {
 
 	e.keyParsed = key.AsString()
 
-	attr = content.Attributes["default"]
+  attr, ok := content.Attributes["default"]
+	if !ok {
+		e.def = ""
+		return diags
+	}
 	key, d = attr.Expr.Value(hclContext)
 	diags = diags.Extend(d)
 

--- a/tests/tests/failing/env-data-key-missing/togomak.hcl
+++ b/tests/tests/failing/env-data-key-missing/togomak.hcl
@@ -1,0 +1,11 @@
+togomak {
+  version = 1
+}
+
+data "env" "hello" {
+}
+
+stage "example" {
+  name   = "example"
+  script = "echo hello world ${data.env.hello.value}"
+}

